### DISCUSSION TITLE
Specifying webpack configuration using node v23 in next config breaks next build

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#72868] [Help]
### Link to the code that reproduces this issue

https://github.com/dualdetail/webpack-example

### To Reproduce

npm install
npm run build

![Screenshot 2024-10-26 212121](https://github.com/user-attachments/assets/e1cfe7e7-aa65-4220-b719-b7cc3c43d846)

### Current vs. Expected behavior

Currently, build prints webpack caching failure then stops building (this also prints in dev without issues). If you remove the webpack configuration from next.config.ts the build will process normally. I set it up so the next.config.ts file does not change the config but just returns the default which still fails.

Expected behavior is build will work with webpack configuration.

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Pro
  Available memory (MB): 98051
  Available CPU cores: 32
Binaries:
  Node: 23.0.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 15.0.2-canary.8 // Latest available version is detected (15.0.2-canary.8).
  eslint-config-next: 15.0.2-canary.8
  react: 19.0.0-rc-1631855f-20241023
  react-dom: 19.0.0-rc-1631855f-20241023
  typescript: 5.6.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Webpack

### Which stage(s) are affected? (Select all that apply)

next build (local)

### Additional context

I have tried using both npm and pnpm without success. I did not get webpack.cache.PackFileCacheStrategy print out in nextjs 14.2.15 and node v23.

**Comment by ghost** [comment]
So I experimented on accident. It seems related to node v23. I was able to build the app using node v22 just fine. I tested this on a separate machine and updating to node v23 and installing libraries produces the same output (as in it doesn't build)

**Comment by EugeCos** [comment]
Similar problem here, build is breaking with below error after bumping Node to 23.1.0.
```
Error: Could not find a production build in the '.next' directory. Try building your app with 'next build' before starting the production server. https://nextjs.org/docs/messages/production-start-no-build-id
```

**Comment by Plow74** [comment]
Having this issue as well building a Next 13.5.6 app on Node v23.

`Creating an optimized production build  .<w> [webpack.cache.PackFileCacheStrategy] Caching failed for pack: Error: Unable to snapshot resolve dependencies`

_Edit by maintainer bot: Comment was **automatically** minimized because it was considered unhelpful. (If you think this was by mistake, let us know). Please only comment if it adds context to the issue. If you want to express that you have the same problem, use the upvote 👍 on the issue description or subscribe to the issue for updates. Thanks!_

**Comment by gyerzh** [comment]
same issue for me for node:23.1.0 - <w> [webpack.cache.PackFileCacheStrategy] Caching failed for pack: Error: Unable to snapshot resolve dependencies

works fine with node:22.6.0

**Comment by MuntasirSZN** [comment]
Same issue

_Edit by maintainer bot: Comment was **automatically** minimized because it was considered unhelpful. (If you think this was by mistake, let us know). Please only comment if it adds context to the issue. If you want to express that you have the same problem, use the upvote 👍 on the issue description or subscribe to the issue for updates. Thanks!_

**Comment by windsornguyen** [comment]
Can confirm Node v23 breaks, but Node **v22.11.0** currently works for me

**Comment by MuntasirSZN** [comment]
> Can confirm Node v23 breaks, but Node **v22.11.0** currently works for me

Yeah. Node 23 breaks, I tested up to 22.11.0, and it works. Maybe node v23 is the culprit of all of this? Me sticking to node v22.11.0

**Comment by AnnatarHe** [comment]
23.2 breaks, 23.1 works for me
